ssg/variables.py

```diff
@@ -0,0 +1,415 @@
+        profile_variables (dict): A dictionary of profile variables to be updated.
+        policies (dict): A dictionary of policies, where each key is a policy ID and each value is
+                         a policy object containing the controls.
+
+    Returns:
+        dict: The updated profile variables dictionary.
+
+    Raises:
+        KeyError: If the policy ID from the control line is not found in the policies dictionary.
+    """
+    if control_line.count(":") == 2:
+        policy_id, _, level = control_line.split(":")
+    else:
+        policy_id, _ = control_line.split(":")
+        level = None
+
+    try:
+        policy = policies[policy_id]
+    except KeyError:
+        print(f"Policy {policy_id} not found")
+        return profile_variables
+
+    for control in policy.controls:
+        if level in control.levels:
+            for rule in control.rules:
+                if "=" in rule:
+                    variable_name, variable_value = rule.split('=', 1)
+                    # When a profile extends a control file, the variables explicitly defined in
+                    # profiles files must be honored, so don't update variables already defined.
+                    if variable_name not in profile_variables:
+                        profile_variables[variable_name] = variable_value
+    return profile_variables
+
+
+def _process_selections(profile_yaml: dict, profile_variables: dict, policies: dict) -> dict:
+    """
+    Processes the selections from the profile YAML and updates the profile variables accordingly.
+
+    Args:
+        profile_yaml (dict): A dictionary containing the profile YAML data.
+        profile_variables (dict): A dictionary to store the profile variables.
+        policies (dict): A dictionary containing policy information.
+
+    Returns:
+        dict: The updated profile variables dictionary.
+    """
+    selections = profile_yaml.get("selections", None)
+    for selected in selections:
+        if "=" in selected and "!" not in selected:
+            variable_name, variable_value = selected.split('=', 1)
+            profile_variables[variable_name] = variable_value
+        elif ":" in selected:
+            profile_variables = _process_controls(selected, profile_variables, policies)
+    return profile_variables
+
+
+def _process_profile(profiles_files: list, file: str, policies: dict, profile_variables=None) -> dict:
+    """
+    Processes a profile by loading its YAML file, handling profile extensions, and processing selections.
+
+    Args:
+        profiles_files (list): A list of profile file paths.
+        file (str): The path to the profile file to be processed.
+        policies (dict): A dictionary of policies defined by control files.
+        profile_variables (dict, optional): A dictionary of profile variables. Defaults to None.
+
+    Returns:
+        dict: A dictionary containing the processed profile variables.
+    """
+    if profile_variables is None:
```

Comment:
I think this condition is not needed if you use {} instead of None as a default value in the function definition.

Reply:
Updated.